sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -76,7 +79,7 @@ class BasicWriteTaskStatsTracker(hadoopConf: Configuration)
 
 
   override def newPartition(partitionValues: InternalRow): Unit = {
-    numPartitions += 1
+    partitions = partitions :+ partitionValues
```

Comment:
this looks like appending a immutable collection. Can we be more explicit? e.g. `partitions.append(partitionValues)`

Reply:
`partitions.appended(partitionValues)` need Scala 2.13

Reply:
or `add`? doesn't scala 2.12 have java-like method to add elements?

Reply:
Address 6d80788 change to use `ArrayBuffer` to store `partitions ` and there change to `partitions.appended(partitionValues)`